### PR TITLE
Feature/d3 asim 3633

### PR DESCRIPTION
**Pull request by spyrostz**
- Removed deprecated parameter SELL_ON_MOST_EXPENSIVE_MARKET
- Adapted generate_market_slot_list method to include the future market time slots, in order to correctly expand the profiles to take into account future markets too.

### CODE REVIEW

**Review by Hassan754 — APPROVED** [review]
LGTM

**Review by Hassan754 — APPROVED** [review]

File: tests/test_user_profile.py [hunk 1]
```diff
@@ -91,3 +95,30 @@ def test_read_profile_for_player(self):
             datetime(2021, 2, 12, 0, 15, 0): 0.5,
             datetime(2021, 2, 12, 0, 30, 0): 0.1
         })
+
+    def test_correct_time_expansion_read_arbitrary_profile(self):
```
Comment (hannesdiedrich) [inline tests/test_user_profile.py:99]: Following our style convention i would propose:
```suggestion
    def test_read_arbitrary_profile_returns_pofile_with_correct_time_expansion(self):
```

File: tests/test_user_profile.py [hunk 2]
```diff
@@ -91,3 +95,30 @@ def test_read_profile_for_player(self):
...
+            mmr = read_arbitrary_profile(InputProfileTypes.IDENTITY, market_maker_rate)
+            assert list(mmr.keys())[-1] == expected_last_time_slot
+        else:
+            GlobalConfig.FUTURE_MARKET_DURATION_HOURS = 0
+            GlobalConfig.sim_duration = duration(hours=3)
+            mmr = read_arbitrary_profile(InputProfileTypes.IDENTITY, market_maker_rate)
+            assert (list(mmr.keys())[-1] - today(tz=TIME_ZONE)).days == 0
+            GlobalConfig.sim_duration = duration(hours=36)
+            mmr = read_arbitrary_profile(InputProfileTypes.IDENTITY, market_maker_rate)
+            assert (list(mmr.keys())[-1] - today(tz=TIME_ZONE)).days == 1
```
Comment (hannesdiedrich) [inline tests/test_user_profile.py:119]: How about being even more and check for the hours in order to catch (unlikely) errors that only add a few future markets.
```suggestion
            assert (list(mmr.keys())[-1] - today(tz=TIME_ZONE)).hours == 24
```
Also applies to the other checks.

**Review by hannesdiedrich — APPROVED** [review]
Two minor comments. LGTM in general

**Review by faizan2590 — DISMISSED** [review]
LGTM

File: tests/test_user_profile.py (hunk 2 above)
Comment (spyrostz) [reply tests/test_user_profile.py:119]: It does not work like that. Pendulum duration(days=1) has the following members:
duration.hours = 0
duration.days = 1
In order to satisfy it I will save it in a variable and compare both

**Review by Hassan754 — APPROVED** [review]
LGTM

File: tests/test_user_profile.py (hunk 1 above)
Comment (spyrostz) [reply tests/test_user_profile.py]: Forgot to leave a comment here, should be fixed by now !

File: tests/test_user_profile.py (hunk 2 above)
Comment (hannesdiedrich) [reply tests/test_user_profile.py:119]: Darn, I always confuse this. I actually meant:
```suggestion
            assert (list(mmr.keys())[-1] - today(tz=TIME_ZONE)).total_hours == 24
```

**Pull request merged**